nifi-nar-bundles/nifi-mqtt-bundle/nifi-mqtt-processors/src/main/java/org/apache/nifi/processors/mqtt/adapters/HiveMqV5ClientAdapter.java

```diff
@@ -0,0 +1,194 @@
+        Objects.requireNonNull(callback, "callback should be set");
+
+        mqtt5Client.toAsync().subscribeWith()
+                .topicFilter(topicFilter)
+                .qos(Objects.requireNonNull(MqttQos.fromCode(qos)))
+                .callback(mqtt5Publish -> {
+                    final NifiMqttMessage nifiMqttMessage = new NifiMqttMessage();
+                    nifiMqttMessage.setPayload(mqtt5Publish.getPayloadAsBytes());
+                    nifiMqttMessage.setQos(mqtt5Publish.getQos().getCode());
+                    nifiMqttMessage.setRetained(mqtt5Publish.isRetain());
+                    try {
+                        callback.messageArrived(mqtt5Publish.getTopic().toString(), nifiMqttMessage);
+                    } catch (Exception e) {
+                        throw new NifiMqttException(e);
+                    }
+                })
+                .send();
+    }
+
+    @Override
+    public void setCallback(NifiMqttCallback callback) {
+        this.callback = callback;
+    }
+
+    public static KeyManagerFactory getKeyManagerFactory(String keyStoreType, String path, char[] keyStorePassword) {
+        try {
+            final KeyStore keyStore = loadIntoKeyStore(keyStoreType, path, keyStorePassword);
+
+            final KeyManagerFactory kmf = KeyManagerFactory
+                    .getInstance(KeyManagerFactory.getDefaultAlgorithm());
+            kmf.init(keyStore, new char[0]); // https://stackoverflow.com/questions/1814048/sun-java-keymanagerfactory-and-null-passwords
+
+            return kmf;
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public static TrustManagerFactory getTrustManagerFactory(String trustStoreType, String path, char[] keyStorePassword) {
+        try {
+            final KeyStore trustStore = loadIntoKeyStore(trustStoreType, path, keyStorePassword);
+
+            final TrustManagerFactory tmf = TrustManagerFactory
+                    .getInstance(TrustManagerFactory.getDefaultAlgorithm());
+            tmf.init(trustStore);
+
+            return tmf;
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static KeyStore loadIntoKeyStore(String type, String path, char[] keyStorePassword) throws KeyStoreException, IOException, NoSuchAlgorithmException, CertificateException {
+        final KeyStore keyStore = KeyStore.getInstance(type);
+        final InputStream in = new FileInputStream(path);
+        keyStore.load(in, keyStorePassword);
+        return keyStore;
+    }
+
+    private byte[] toBytes(char[] chars) {
+        final ByteBuffer byteBuffer = StandardCharsets.UTF_8.encode(CharBuffer.wrap(chars));
+        final byte[] bytes = Arrays.copyOfRange(byteBuffer.array(), byteBuffer.position(), byteBuffer.limit());
+        clearSensitive(byteBuffer.array());
+        return bytes;
+    }
+
+    private void clearSensitive(char[] chars) {
```

Comment:
Clearing the password from the _MqttConnectionProperties_ prevents the the reconnection logic to execute successfully.
I'd rather remove these _clear_ methods as acquiring the password when you have access to the memory of the nifi process is not preventable like this anyway.

Reply:
Changed, thanks for noticing it.